lib/restbase.js

```diff
@@ -283,8 +282,18 @@ RESTBase.prototype._request = function(req) {
         }
 
         // Call the handler with P.try to catch synchronous exceptions.
-        var reqHandlerPromise = P.try(handler, [childRESTBase, childReq])
-        .then(function(res) {
+        var reqHandlerPromise;
+        if (handler.validator) {
+            reqHandlerPromise = P.try(function() {
+                return handler.validator.validate(childReq);
+            })
+            .then(function() {
+                return handler(childRESTBase, childReq);
```

Comment:
Ah right. From the [BB docs](https://github.com/petkaantonov/bluebird/blob/master/API.md#promisetryfunction-fn--arraydynamicdynamic-arguments--dynamic-ctx----promise):

> Start the chain of promises with `Promise.try`. Any synchronous exceptions will be turned into rejections on the returned promise.